apps/app/src/app/shared/layout/page-nav/page-nav.component.ts

```diff
@@ -15,17 +34,43 @@ import { pageNavTmpl } from '~/app/shared/layout/page-nav/page-nav-outlet.compon
 				<div class="space-y-2 px-1">
 					<h3 class="font-medium">On this page</h3>
 					<ul class="m-0 flex list-none flex-col">
-						<ng-content />
+						@for (link of links(); track link.id) {
+							<spartan-page-nav-link [ngClass]="{ 'pl-4': link.isNested }" [fragment]="link.id" [label]="link.label" />
+						} @empty {
+							@if (isDevMode()) {
+								[DEV] Nothing to see here!
+							}
+						}
 					</ul>
 				</div>
 			</hlm-scroll-area>
 		</ng-template>
 	`,
 })
-export class PageNavComponent implements AfterViewInit, OnDestroy {
+export class PageNavComponent implements OnInit, AfterViewInit, OnDestroy {
 	@ViewChild('pageNav', { static: true })
 	pageNavTpl?: TemplateRef<unknown>;
 
+	isDevMode = signal(isDevMode());
+	links = signal<SamePageAnchorLink[]>([]);
+
+	private page: HTMLElement = (inject(ElementRef).nativeElement as HTMLElement).previousSibling as HTMLElement;
+
+	ngOnInit() {
+		const selectors = [
+			'section[spartanMainSection] spartan-section-sub-heading',
+			'section[spartanMainSection] > h3',
+			'section[spartanMainSection] section > h3',
```

Comment:
Just a minor thing. 
With these selectors you revealed, that the structure of the pages is different in the components, than in the components section. 
Not sure what is the best approach, "with section" , "without section", or a own directive for the h3 that we can select. 
Maybe @goetzrobin can tell his opinion on this.

Reply:
True. IMO, we shouldn't need the last selector, which is used only in `cli.page.ts`. I didn't want to update the structure of the existing content too much, so I added the selector instead.

Having a directive would be interesting. We have `HlmH3Directive` already, which I can apply here (and newer authors would need to remember adding this), or we could apply this directive to `section[spartanMainSection] > h3` as part of the selector.

Reply:
I've added a commit addressing the missing IDs on the h3 tags. I can update this again if we want to use a directive instead for now, though in the long term this would be handled as part of markdown somehow (https://github.com/analogjs/analog/issues/448)